app/src/main/java/couclou/fr/nixconfighelper/MainAdapter.kt

```diff
@@ -25,24 +39,126 @@ class MainAdapter(val feed:Feed): RecyclerView.Adapter<CustomViewHolder>() {
             val readOnly = element.readOnly
             val defaultValue = element.defaultValue
             var description = element.description.replace("\n","")
-
-            holder.view.keynameValue.text = keyname
-            holder.view.typeValue.text = type
-            holder.view.readOnlyValue.text = readOnly.toString()
-            holder.view.defaultValueValue.text = defaultValue
-            holder.view.descriptionValue.text = description
-
             var declarations: String = ""
-            if (declarations.isNotEmpty()) {
+            if (element.declarations.isNotEmpty()) {
                 for (declaration in element.declarations) {
-                    declarations = "$declarations \n $declaration"
+                    declarations = "$declarations\n$declaration"
                 }
             }
+
+            holder.view.option_keynameValue.text = keyname
+            holder.view.typeValue.text = type
+            holder.view.readOnlyValue.text = readOnly.toString()
+            holder.view.defaultValueValue.text = defaultValue
+            holder.view.option_descriptionValue.text = description
             holder.view.declarationsValue.text = declarations
         }
 
         else if (feed.elements.get(position) is Package) {
             // handle package element type
+            val element = (feed.elements.get(position) as Package)
+
+            val id = element.id
+            val channel = element.channel
+            val keyname = element.keyname
+            val name = element.name
+            val pname = element.pname
+            val version = element.version
+            val system = element.system
+            val available = element.available
+            val description = element.description
+            val downloadPage = element.downloadPage
+            val homepage = element.homepage
+            val license = "Full name : ${element.license.fullName}\nShort name : ${element.license.shortName}\n" +
```

Comment:
https://discuss.kotlinlang.org/t/java-string-format-alternative-in-kotlin/703/4

On évite en général d'injecter des variables à la volée comme ça, c'est pas super lisible